packages/bot-web-ui/src/stores/quick-strategy-store-1.ts

```diff
@@ -0,0 +1,140 @@
+    setValue: (name: string, value: string | number) => void;
+    onSubmit: (data: TFormData, run: boolean) => void;
+}
+
+export default class QuickStrategyStore implements IQuickStrategyStore {
+    root_store: RootStore;
+    is_open = false;
+    selected_strategy = 'MARTINGALE';
+    form_data: { [key: string]: string | number } = {
+        symbol: '1HZ100V',
+        trade_type: 'callput',
+        duration_unit: 't',
+    };
+
+    constructor(root_store: RootStore) {
+        makeObservable(this, {
+            is_open: observable,
+            selected_strategy: observable,
+            form_data: observable,
+            setFormVisibility: action,
+            setSelectedStrategy: action,
+            onSubmit: action,
+        });
+        this.root_store = root_store;
+        reaction(
+            () => this.is_open,
+            () => {
+                if (!this.is_open) {
+                    this.selected_strategy = 'MARTINGALE';
+                }
+            }
+        );
+    }
+
+    setFormVisibility = (is_open: boolean) => {
+        this.is_open = is_open;
+    };
+
+    setSelectedStrategy = (strategy: string) => {
+        this.selected_strategy = strategy;
+    };
+
+    setValue = (name: string, value: string | number) => {
+        this.form_data[name] = value;
+    };
+
+    onSubmit = async (data: TFormData, run: boolean) => {
+        const { contracts_for } = ApiHelpers.instance;
+        const market = await contracts_for.getMarketBySymbol(data.symbol);
+        const submarket = await contracts_for.getSubmarketBySymbol(data.symbol);
+        const trade_type_cat = await contracts_for.getTradeTypeCategoryByTradeType(data.trade_type);
+        const selected_strategy = STRATEGIES[this.selected_strategy];
+        const strategy_xml = await import(/* webpackChunkName: `[request]` */ `../xml/${selected_strategy.name}.xml`);
+        const strategy_dom = Blockly.Xml.textToDom(strategy_xml.default);
+        const modifyValueInputs = (key: string, value: number) => {
+            const el_value_inputs = strategy_dom?.querySelectorAll(`value[strategy_value="${key}"]`);
+
+            el_value_inputs?.forEach((el_value_input: HTMLElement) => {
+                el_value_input.innerHTML = `<shadow type="math_number"><field name="NUM">${value}</field></shadow>`;
+            });
+        };
+
+        const modifyFieldDropdownValues = (name: string, value: string) => {
+            const name_list = `${name.toUpperCase()}_LIST`;
+            const el_blocks = strategy_dom?.querySelectorAll(`field[name="${name_list}"]`);
+
+            el_blocks?.forEach((el_block: HTMLElement) => {
+                el_block.innerHTML = value;
+            });
+        };
+        const { unit, ...rest_data } = data;
+        const fields_to_update = {
+            market,
+            submarket,
+            tradetypecat: trade_type_cat,
+            alembert_unit: unit,
+            oscar_unit: unit,
```

Comment:
@sandeep-deriv Bhai for these two, can we update the XML itself to take unit instead of `alembert_unit` or `oscar_unit`? Otherwise later if there are more strategies with their specific names in the XML then we always have to touch this function.

Reply:
this we need to check with Riordan and maybe later we will keep one filed in all of the strategies @shafin-deriv